rules/internal/utils/extension.go

```diff
@@ -147,3 +147,22 @@ func GetResourceReference(f *desc.FieldDescriptor) *apb.ResourceReference {
 	}
 	return nil
 }
+
+// FindResource attempts to find the google.api.resource(_definition) given the
```

Comment:
Returns first resource of type matching the reference param.

Reply:
Done.